src/shogun2-util/src/main/java/de/terrestris/shogun2/util/naming/PhysicalNamingStrategyShogun2.java

```diff
@@ -0,0 +1,131 @@
+package de.terrestris.shogun2.util.naming;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hibernate.boot.model.naming.Identifier;
+import org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+/**
+ * Limits identifier length if necessary.
+ *
+ * @author Nils Bühner
+ *
+ */
+public class PhysicalNamingStrategyShogun2 extends PhysicalNamingStrategyStandardImpl {
+
+	private static final long serialVersionUID = 1L;
+
+	protected static final int LENGTH_LIMIT_ORACLE = 30;
+
+	protected static final int LENGTH_LIMIT_POSTGRESQL = 63;
+
+	@Autowired(required = false)
+	@Qualifier("tablePrefix")
+	private String tablePrefix;
```

Comment:
At first, the `@Autowired` annotation is configured with `required=false`, i.e. the prefix value is not mandatory.

`private String tablePrefix;` is the equal to `private String tablePrefix = null;` as `null` is the default value.

On top of that, the prefix value will only be used if it is not null:

https://github.com/buehner/shogun2/blob/naming-strategy/src/shogun2-util/src/main/java/de/terrestris/shogun2/util/naming/PhysicalNamingStrategyShogun2.java#L69-L71